dnf/base.py

```diff
@@ -1746,7 +1748,10 @@ def _install_multiarch(self, query, reponame=None, strict=True):
         self._report_already_installed(already_inst)
         for a in available:
             sltr = dnf.selector.Selector(self.sack)
-            sltr = sltr.set(pkg=a)
+            q = self.sack.query().filterm(pkg=a)
+            if self.conf.obsoletes:
+                q = q.union(self.sack.query().filterm(obsoletes=q))
+            sltr = sltr.set(pkg=q)
```

Comment:
What about?
```
            if self.conf.obsoletes:
                a = self.sack.query().filterm(pkg=a)
                q = a.union(self.sack.query().filterm(obsoletes=a))
            sltr = sltr.set(pkg=a)
```
I will not create a query when  self.conf.obsoletes is False?

Also I would recommend to rename variable `a` as for example `packages_or_query` (Not nice I know)

Reply:
Systems with conf.obsoletes=False are very rare so the benefit would be really small. I agree with renaming the `a` variable, but do not like much that it should contain either list or query. I'd rather go with my original solution where query and packages list stay separated.